.environment/terraform-fmt/run-terraform-fmt.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+function error() {
+    echo "terraform-fmt> ERROR: ${*}"
+}
+
+function warning() {
+    echo "terraform-fmt> Warning: ${*}"
+}
+
+function note() {
+    echo "terraform-fmt> info: ${*}"
+}
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+TERRAFORM_NEEDS_FMT=0
+LOGFILE="terraform-fmt.log"
+
+function terraform_check_fmt() {
+    note "Checking Terraform formatting."
+    make -C "${REPO_ROOT?}/operations" -f "${REPO_ROOT?}/operations/Makefile" tf-cmd TF_CMD="terraform fmt -check -recursive /app/src" > "${REPO_ROOT?}/${LOGFILE?}" 2>&1
+    TERRAFORM_NEEDS_FMT=$?
+    return ${TERRAFORM_NEEDS_FMT?}
+}
+
+function terraform_fmt() {
+    if [[ ${TERRAFORM_NEEDS_FMT?} != 0 ]]; then
+        warning "Formatting all Terraform files."
+        make -C "${REPO_ROOT?}/operations" -f "${REPO_ROOT?}/operations/Makefile" tf-cmd TF_CMD="terraform fmt -recursive /app/src" >> "${REPO_ROOT?}/${LOGFILE?}" 2>&1
+        return $?
+    fi
+    return 0
+}
+
+terraform_check_fmt
+RC=$?
+
+terraform_fmt
+
+if [[ ${RC?} != 0 ]]; then
+    error "(return code=${RC?}) Your Terraform files are not formatted."
+    error "\`terraform fmt\` has been run and the changes can be committed to comply with formatting requirements."
```

Comment:
Any changes performed by `terraform fmt` are not going to be staged and so must be added to the user's staged changes:

```suggestion
    error "\`terraform fmt\` has been run and the changes can be committed to comply with formatting requirements."
    error "Please validate the output from terraform fmt and then git add it to your staged changes."
```

There's a subtle bug here: if I commit non-compliant code that triggers this, it will format my code thus making it compliant. But, the formatted - and thus compliant - code is not yet staged. However, when I try to recommit - without doing _any_thing else, the hook will now pass because it checks the content of the files listed in your staging as it is in your working directory.

So I think it may make more sense to NOT try to format for the user and tell them to do the formatting (and then any subsequently required staging/git adding) themselves.

My suggestion would thus actually be to NOT try to format for the user!

Reply:
Agreed. Updated the hook to not automatically format.